src/app/webbrowser/TabsList.qml

```diff
@@ -18,26 +18,39 @@
 
 import QtQuick 2.4
 import Ubuntu.Components 1.3
+import QtQml.Models 2.2
 
 Item {
     id: tabslist
 
     property real delegateHeight
     property real chromeHeight
-    property alias model: repeater.model
-    readonly property int count: repeater.count
+    property real tabChromeHeight: units.gu(5)
+    property alias model: filteredModel.model
+    readonly property int count: model.count
+    property alias searchText: searchField.text
+    property alias view: list.item
     property bool incognito
+    property bool searchMode: false
 
     signal scheduleTabSwitch(int index)
     signal tabSelected(int index)
     signal tabClosed(int index)
 
     function reset() {
-        flickable.contentY = 0
+        tabslist.view.contentY = 0
+        searchText = ""
     }
 
     readonly property bool animating: selectedAnimation.running
 
+    Connections {
+        // WORKAROUND: Repeater items in listNarrowComponent stay hidden when switching from wide to narrow layout
+        // if the model is direcly assigned in its definition. This solves that issue.
+        target: browser
+        onWideChanged: if (!target.wide) searchText = " "
```

Comment:
Is setting the search text to `" "` different than setting it to `""`?

Reply:
yes, `""` won't fix the issue :sweat_smile: 
It's not a good workaround, I admit but that's what I came up after hours of debugging :)
Maybe I can go back in the future and see if I can find an actual fix.